src/main/java/com/marginallyclever/makelangelo/makeArt/io/vector/LoadSVG.java

```diff
@@ -239,112 +240,111 @@ private void arcTurtle(Turtle turtle,double cx,double cy,double rx,double ry,dou
 			Element element = (Element)node.item( iPathNode );
-			if(isElementStrokeNone(element)) 
+			if(isElementStrokeNone(element))
 				continue;
-			
+
 			Matrix3d m = getMatrixFromElement(element);
-			
+
 			double cx=0,cy=0,r=0;
 			if(element.hasAttribute("cx")) cx = Double.parseDouble(element.getAttribute("cx"));
 			if(element.hasAttribute("cy")) cy = Double.parseDouble(element.getAttribute("cy"));
 			if(element.hasAttribute("r" )) r  = Double.parseDouble(element.getAttribute("r"));
 			v2 = transform(cx+r,cy,m);
 			myTurtle.jumpTo(v2.x,v2.y);
-			
+
 			double circ = Math.PI * 2.0 * r;
 			circ = Math.ceil(Math.min(Math.max(3,circ),360));
-			
-		    logger.debug("circ={}", circ);
-			for(double i=1;i<circ;++i) {
-				double v = (Math.PI*2.0) * (i/circ);
-				double s=r*Math.sin(v);
-				double c=r*Math.cos(v);
-				v2 = transform(cx+c,cy+s,m);
-				myTurtle.moveTo(v2.x,v2.y);
-			}
-			v2 = transform(cx+r,cy,m);
-			myTurtle.moveTo(v2.x,v2.y);
-	    }
+
+			logger.debug("circ={}", circ);
+			printEllipse(m, cx, cy, r, r, circ);
+		}
 	}
 
 	private void parseEllipseElements(NodeList node) {
 		Vector3d v2;
-		
-	    int pathNodeCount = node.getLength();
-	    for( int iPathNode = 0; iPathNode < pathNodeCount; iPathNode++ ) {
+
+		int pathNodeCount = node.getLength();
+		for( int iPathNode = 0; iPathNode < pathNodeCount; iPathNode++ ) {
 			Element element = (Element)node.item( iPathNode );
-			if(isElementStrokeNone(element)) 
+			if(isElementStrokeNone(element))
 				continue;
-			
+
 			Matrix3d m = getMatrixFromElement(element);
-			
+
 			double cx=0,cy=0,rx=0,ry=0;
 			if(element.hasAttribute("cx")) cx = Double.parseDouble(element.getAttribute("cx"));
 			if(element.hasAttribute("cy")) cy = Double.parseDouble(element.getAttribute("cy"));
 			if(element.hasAttribute("rx")) rx = Double.parseDouble(element.getAttribute("rx"));
 			if(element.hasAttribute("ry")) ry = Double.parseDouble(element.getAttribute("ry"));
 			v2 = transform(cx+rx,cy,m);
 			myTurtle.jumpTo(v2.x,v2.y);
-			
-			double circ = Math.min(3,Math.floor(Math.PI * ry*rx)); 
-			for(double i=1;i<circ;++i) {
-				double v = (Math.PI*2.0) * (i/circ);
-				double s=ry*Math.sin(v);
-				double c=rx*Math.cos(v);
-				v2 = transform(cx+c,cy+s,m);
-				myTurtle.moveTo(v2.x,v2.y);
-			}
-			v2 = transform(cx+rx,cy,m);
+
+			double perimeterOfAnEllipseApprox = Math.PI * 2.0 * Math.sqrt((ry*ry + rx*rx)/2.0);
+			double steps = Math.max(3,perimeterOfAnEllipseApprox);
+			steps = Math.min(60,steps);
+			printEllipse(m, cx, cy, rx, ry, perimeterOfAnEllipseApprox);
+		}
+	}
+
+	private void printEllipse(Matrix3d m, double cx, double cy, double rx, double ry, double perimeterOfAnEllipseApprox) {
+		Vector3d v2;
+		for(double i = 1; i<perimeterOfAnEllipseApprox; ++i) {
+			double v = (Math.PI*2.0) * (i/perimeterOfAnEllipseApprox);
+			double s=ry*Math.sin(v);
+			double c=rx*Math.cos(v);
+			v2 = transform(cx+c,cy+s,m);
 			myTurtle.moveTo(v2.x,v2.y);
-	    }
+		}
```

Comment:
Just to be sure to understand it right (sorry long discussion)

m is a transformation matrix (x,y translation and rotation (no scale))

we draw here small segments of 1 unit (i++) (assuming that the unit/granule is 1mm (before any other turtle transformation))

whatever size our circle/ellipse is, we should have a "segment" resolution/granul of ~1?

otherwise, the segments are drawn inside the arc (only their endpoints fall exactly on the path of the arc)?

We are in the case circle (green) our aproximation is the bleu ?
![image](https://user-images.githubusercontent.com/94939582/154948480-29d1f608-4292-4f63-9217-19e3420b7d91.png)



from https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/Primitive_Solids#sphere 

because circular paths are approximated with polygons inscribed within in a circle. The points of the polygon are on the circle, but straight lines between are inside. To have all of the hole larger than the true circle, the polygon must lie wholly outside of the circle (circumscribed). [Modules for circumscribed holes](https://en.wikibooks.org/wiki/OpenSCAD_User_Manual/undersized_circular_objects)

Reply:
Yes, circles become slightly undersized.  Same problem on 3D printers.  the maximum number of segments in a circle is capped.  Above a certain amount it doesn't matter, people can't tell the difference.  ironically it's the smaller circles with too many instructions that make the machine run poorly.  Gcode has G02 and G03 arc commands, but Turtle does not support arcs because it creates A LOT of work for us.